ci/build.yml

```diff
@@ -0,0 +1,21 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.17
```

Comment:
I think we should be using `1.17.6` here (see https://github.com/ONSdigital/dp/blob/main/standards/DEPENDENCY_UPGRADING.md#build-and-test-version-docker-image)

Also in unit.yml and component.yml